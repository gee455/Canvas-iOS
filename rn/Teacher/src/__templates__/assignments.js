//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

/* @flow */

import template, { type Template } from '../utils/template'

export const assignment: Template<Assignment> = template({
  id: '123456789',
  course_id: '987654321',
  name: 'learn to write code',
  description: 'a course that can take any beginner to professional in a matter of microseconds.  Harness the power of your favorite languages like PHP and Cold Fusion.',
  due_at: '2037-06-01T05:59:00Z',
  unlock_at: '2017-03-17T06:00:00Z',
  lock_at: '2037-06-01T05:59:59Z',
  created_at: '2017-03-17T19:15:25Z',
  updated_at: '2017-03-17T19:15:25Z',
  points_possible: 20,
  grading_type: 'points',
  assignment_group_id: 1376,
  peer_reviews: false,
  automatic_peer_reviews: false,
  position: 1,
  submission_types:
  [ 'online_text_entry',
    'online_url',
    'media_recording',
    'online_upload' ],
  needs_grading_count: 0,
  html_url: 'http://hostname.tld/whatever',
  overrides: [],
  grade_group_students_individually: false,
  group_category_id: null,
  has_overrides: false,
  only_visible_to_overrides: false,
  published: true,
  unpublishable: false,
  rubric_assessment: {},
  rubric_settings: null,
  rubric: null,
})

export const assignmentGroup: Template<AssignmentGroup> = template({
  id: '1',
  name: 'Learn React Native',
  position: 1,
  assignments: [assignment()],
  group_weight: 1,
  integration_data: {},
  sis_source_id: '',
})

export const assignmentDueDate: Template<AssignmentDate> = template({
  title: 'due',
  due_at: '2037-06-01T05:59:00Z',
  unlock_at: '2017-03-17T06:00:00Z',
  lock_at: '2037-06-01T05:59:59Z',
})

export const assignmentOverride: Template<AssignmentOverride> = template({
  id: '1',
  assignment_id: '1',
  student_ids: [],
  group_id: '1',
  course_section_id: '1',
  title: 'override',
  all_day: false,
  all_date_date: '',
  due_at: '2037-06-01T05:59:00Z',
  unlock_at: '2017-03-17T06:00:00Z',
  lock_at: '2037-06-01T05:59:59Z',
})
