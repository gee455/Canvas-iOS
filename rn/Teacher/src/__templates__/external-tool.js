//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

/* @flow */

import template, { type Template } from '../utils/template'

export const externalTool: Template<AddressBookResult> = template({
  id: '1',
  name: 'LTI FTW',
  url: 'https://canvas.instructure.com/lti',
})

export const ltiLaunchDefinition: Template<LtiLaunchDefinition> = template({
  definition_id: 42,
  placements: {
    course_navigation: {
      url: 'https://rollcall.instructure.com/launchy-launch',
    },
  },
})
