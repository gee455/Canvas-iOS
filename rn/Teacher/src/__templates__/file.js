//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

/* @flow */

import template, { type Template } from '../utils/template'

export const usageRights: Template<UsageRights> = template({
  legal_copyright: '',
  use_justification: 'own_copyright',
})

export const file: Template<File> = template({
  id: '111',
  display_name: 'Book Report',
  url: 'http://canvaslms.com/bookreport',
  size: 488212,
  thumbnail_url: 'http://fillmurray.com/322/200',
  mime_class: 'document',
  preview_url: '/',
  filename: '/bookreport',
  parent_folder_id: null,
  locked: false,
  hidden: false,
  unlock_at: null,
  lock_at: null,
  usage_rights: usageRights(),
  'content-type': 'application/document',
})
