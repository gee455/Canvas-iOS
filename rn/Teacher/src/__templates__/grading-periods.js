//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

/* @flow */

import template, { type Template } from '../utils/template'

export const gradingPeriod: Template<GradingPeriod> = template({
  id: '1023',
  title: 'First Block',
  start_date: '2014-01-07T15:04:00Z',
  end_date: '2014-05-07T17:07:00Z',
  close_date: '2014-06-07T17:07:00Z',
  weight: 33.33,
})
