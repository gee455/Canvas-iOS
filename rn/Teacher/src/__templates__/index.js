//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow

export * from '../redux/__templates__/app-state'
export * from './account-notification'
export * from './addressBook'
export * from './api'
export * from './assignments'
export * from './attachment'
export * from './conversations'
export * from './course'
export * from './discussion'
export * from './enrollments'
export * from './error'
export * from './external-tool'
export * from './file'
export * from './folder'
export * from './grading-periods'
export * from './group'
export * from './helm'
export * from './launch-definitions'
export * from './mediaComment'
export * from './page'
export * from './quiz'
export * from './quizSubmission'
export * from './role'
export * from './rubric'
export * from './section'
export * from './session'
export * from './submissions'
export * from './tab'
export * from './toDo'
export * from './users'
