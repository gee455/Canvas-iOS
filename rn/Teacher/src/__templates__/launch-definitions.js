//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow

import template, { type Template } from '../utils/template'

export const launchDefinitionGlobalNavigationItem: Template<ExternalToolLaunchDefinitionGlobalNavigationItem> = template({
  message_type: 'basic-lti-launch-request',
  url: 'https://mobiledev.beta.instructuremedia.com/lti/launch?custom_gauge_launch_type=global_nav',
  title: 'Gauge',
})

export const launchDefinition: Template<ExternalToolLaunchDefinition> = template({
  definition_type: 'ContextExternalTool',
  definition_id: '360860',
  name: 'Gauge',
  description: 'Tools for Education',
  domain: 'gauge.instructure.com',
  placements: {
    global_navigation: launchDefinitionGlobalNavigationItem(),
  },
})
