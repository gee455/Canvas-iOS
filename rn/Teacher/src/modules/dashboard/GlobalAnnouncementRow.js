//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow

import i18n from 'format-message'
import React from 'react'
import {
  AccessibilityInfo,
  findNodeHandle,
  Image,
  StyleSheet,
  TouchableHighlight,
  View,
} from 'react-native'
import images from '../../images'
import colors from '../../common/colors'
import {
  Text,
  SubTitle,
} from '../../common/text'
import { LinkButton } from '../../common/buttons'
import CanvasWebView from '../../common/components/CanvasWebView'
import DashboardContent from './DashboardContent'

const MESSAGE_STYLE = `
<style>
  html, body {
    color: ${colors.darkText};
    font-size: 16px;
  }
</style>
`

type Props = NavigationProps & {
  style?: any,
  notification: AccountNotification,
  onDismiss: (string) => any,
}

type State = {
  collapsed: boolean,
}

export default class GlobalAnnouncementRow extends React.Component<Props, State> {
  state = {
    collapsed: true,
  }

  content: TouchableHighlight | View | null

  toggle = () => this.setState({
    collapsed: !this.state.collapsed,
  }, () => {
    const handle = findNodeHandle(this.content)
    if (handle) {
      AccessibilityInfo.setAccessibilityFocus(handle)
    }
  })

  dismiss = () => {
    this.props.onDismiss(this.props.notification.id)
  }

  getColorFromIcon (icon: AccountNotificationIcon) {
    switch (icon) {
      case 'error':
        return colors.errorAnnouncementBg
      case 'warning':
        return colors.warningAnnouncementBg
      case 'information':
      default:
        return colors.primaryBrandColor
    }
  }

  getIconSource (icon: AccountNotificationIcon) {
    switch (icon) {
      case 'error':
      case 'warning':
        return images.dashboard.warning
      case 'question':
        return images.dashboard.help
      case 'calendar':
        return images.dashboard.calendar
      case 'information':
      default:
        return images.dashboard.info
    }
  }

  render () {
    const { style, notification: { subject, message, icon } } = this.props
    const { collapsed } = this.state
    const color = this.getColorFromIcon(icon)
    return (
      <DashboardContent
        style={style}
        contentStyle={[{ borderColor: color, borderWidth: 1 }]}
        hideShadow={true}
      >
        {collapsed ? (
          <TouchableHighlight
            accessibilityTraits='button'
            onPress={this.toggle}
            underlayColor='transparent'
            testID='global-announcement-row.expand'
            ref={(node) => { this.content = node }}
          >
            <View style={styles.rowContent}>
              <View style={[styles.iconContainer, { backgroundColor: color }]}>
                <Image source={this.getIconSource(icon)} style={styles.icon} />
              </View>
              <View style={styles.announcementDetails}>
                <Text style={styles.title}>{subject}</Text>
                <SubTitle>{i18n('Tap to view announcement')}</SubTitle>
              </View>
            </View>
          </TouchableHighlight>
        ) : (
          <View style={styles.rowContent}>
            <View style={[styles.iconContainer, { backgroundColor: color }]}>
              <Image source={this.getIconSource(icon)} style={styles.icon} />
            </View>
            <View style={styles.announcementDetails}>
              <View ref={(node) => { this.content = node }}>
                <Text style={styles.openTitle}>{subject}</Text>
              </View>
              <CanvasWebView
                automaticallySetHeight
                style={styles.message}
                html={MESSAGE_STYLE + message}
                navigator={this.props.navigator}
              />
              <LinkButton
                accessibilityLabel={i18n('Dismiss {name}', { name: subject })}
                style={styles.dismiss}
                textStyle={styles.dismissText}
                onPress={this.dismiss}
                testID='global-announcement-row.dismiss'
              >
                {i18n('Dismiss')}
              </LinkButton>
            </View>
          </View>
        )}
        <TouchableHighlight
          accessibilityTraits='button'
          accessibilityLabel={collapsed
            ? i18n('Dismiss {name}', { name: subject })
            : i18n('Hide content for {name}', { name: subject })
          }
          onPress={collapsed ? this.dismiss : this.toggle}
          underlayColor='transparent'
          style={styles.action}
          testID='global-announcement-row.button'
        >
          {collapsed
            ? <Image source={images.x} style={styles.dismissIcon} />
            : <Image source={images.chevronUp} style={styles.collapseIcon} />
          }
        </TouchableHighlight>
      </DashboardContent>
    )
  }
}

const styles = StyleSheet.create({
  icon: {
    marginTop: 14,
    tintColor: 'white',
  },
  rowContent: {
    flexDirection: 'row',
  },
  iconContainer: {
    width: 40,
    alignItems: 'center',
  },
  announcementDetails: {
    marginHorizontal: 12,
    marginVertical: 8,
    flex: 1,
  },
  title: {
    fontWeight: '600',
    marginRight: 32,
  },
  openTitle: {
    fontWeight: '600',
    marginRight: 32,
    marginTop: 8,
  },
  message: {
    marginVertical: 8,
  },
  action: {
    position: 'absolute',
    right: 0,
    top: 0,
    padding: 12,
  },
  dismiss: {
    alignSelf: 'flex-end',
    paddingHorizontal: 4,
    paddingVertical: 2,
  },
  dismissText: {
    fontSize: 16,
  },
  dismissIcon: {
    width: 16,
    tintColor: colors.grey4,
  },
  collapseIcon: {
    marginVertical: 3,
    width: 16,
    tintColor: colors.grey4,
  },
})
