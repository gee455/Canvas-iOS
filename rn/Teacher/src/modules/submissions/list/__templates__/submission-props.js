//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow

import template, { type Template } from '../../../../utils/template'
import type { SubmissionDataProps } from '../submission-prop-types'

const templates = {
  ...require('../../../../__templates__/submissions'),
}

const submissionWithHistory = templates.submissionHistory()

// $FlowFixMe
export const submissionProps: Template<SubmissionDataProps> = template({
  grade: 'B-',
  name: submissionWithHistory.user.name,
  status: 'submitted',
  userID: submissionWithHistory.user.id,
  avatarURL: submissionWithHistory.user.avatar_url,
  sectionID: '1',
})
