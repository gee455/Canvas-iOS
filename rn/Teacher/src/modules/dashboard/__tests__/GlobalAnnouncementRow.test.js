//
// Copyright (C) 2016-present Instructure, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

// @flow

import { shallow } from 'enzyme'
import React from 'react'
import { accountNotification } from '../../../__templates__/account-notification'
import { navigator } from '../../../__templates__/helm'
import GlobalAnnouncementRow from '../GlobalAnnouncementRow'

describe('GlobalAnnouncementRow', () => {
  const defaults = {
    notification: accountNotification({}),
    onDismiss () {},
    navigator: navigator(),
  }

  it('renders error announcements', () => {
    const tree = shallow(
      <GlobalAnnouncementRow
        {...defaults}
        notification={accountNotification({ icon: 'error' })}
      />
    )
    expect(tree).toMatchSnapshot()
  })

  it('renders calendar announcements', () => {
    const tree = shallow(
      <GlobalAnnouncementRow
        {...defaults}
        notification={accountNotification({ icon: 'calendar' })}
      />
    )
    expect(tree).toMatchSnapshot()
  })

  it('renders warning announcements', () => {
    const tree = shallow(
      <GlobalAnnouncementRow
        {...defaults}
        notification={accountNotification({ icon: 'warning' })}
      />
    )
    expect(tree).toMatchSnapshot()
  })

  it('renders question announcements', () => {
    const tree = shallow(
      <GlobalAnnouncementRow
        {...defaults}
        notification={accountNotification({ icon: 'question' })}
      />
    )
    expect(tree).toMatchSnapshot()
  })

  it('renders information announcements', () => {
    const tree = shallow(
      <GlobalAnnouncementRow
        {...defaults}
        notification={accountNotification({ icon: 'information' })}
      />
    )
    expect(tree).toMatchSnapshot()
  })

  it('renders unexpected icon as information', () => {
    const tree = shallow(
      <GlobalAnnouncementRow
        {...defaults}
        notification={accountNotification({ icon: 'bogus' })}
      />
    )
    expect(tree).toMatchSnapshot()
  })

  it('can dismiss the notification', () => {
    const onDismiss = jest.fn()
    const tree = shallow(
      <GlobalAnnouncementRow
        {...defaults}
        notification={accountNotification({ id: '34' })}
        onDismiss={onDismiss}
      />
    )
    tree.find('[testID="global-announcement-row.button"]').simulate('Press')
    expect(onDismiss).toHaveBeenCalledWith('34')
    tree.find('[testID="global-announcement-row.expand"]').simulate('Press')
    tree.find('[testID="global-announcement-row.dismiss"]').simulate('Press')
    expect(onDismiss).toHaveBeenCalledTimes(2)
  })
})
