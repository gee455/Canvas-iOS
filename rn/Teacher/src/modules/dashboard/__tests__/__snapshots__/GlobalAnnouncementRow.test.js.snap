// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalAnnouncementRow renders calendar announcements 1`] = `
<Component
  contentStyle={
    Array [
      Object {
        "borderColor": "#374A59",
        "borderWidth": 1,
      },
    ]
  }
  hideShadow={true}
>
  <TouchableHighlight
    accessibilityTraits="button"
    activeOpacity={0.85}
    onPress={[Function]}
    testID="global-announcement-row.expand"
    underlayColor="transparent"
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "width": 40,
            },
            Object {
              "backgroundColor": "#374A59",
            },
          ]
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/images/dashboard/calendar.png",
            }
          }
          style={
            Object {
              "marginTop": 14,
              "tintColor": "white",
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "marginHorizontal": 12,
            "marginVertical": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "fontWeight": "600",
              "marginRight": 32,
            }
          }
        >
          Permission Slips!
        </Text>
        <SubTitle>
          Tap to view announcement
        </SubTitle>
      </View>
    </View>
  </TouchableHighlight>
  <TouchableHighlight
    accessibilityLabel="Dismiss Permission Slips!"
    accessibilityTraits="button"
    activeOpacity={0.85}
    onPress={[Function]}
    style={
      Object {
        "padding": 12,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    testID="global-announcement-row.button"
    underlayColor="transparent"
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/images/x-icon.png",
        }
      }
      style={
        Object {
          "tintColor": "#8B969E",
          "width": 16,
        }
      }
    />
  </TouchableHighlight>
</Component>
`;

exports[`GlobalAnnouncementRow renders error announcements 1`] = `
<Component
  contentStyle={
    Array [
      Object {
        "borderColor": "#EE0612",
        "borderWidth": 1,
      },
    ]
  }
  hideShadow={true}
>
  <TouchableHighlight
    accessibilityTraits="button"
    activeOpacity={0.85}
    onPress={[Function]}
    testID="global-announcement-row.expand"
    underlayColor="transparent"
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "width": 40,
            },
            Object {
              "backgroundColor": "#EE0612",
            },
          ]
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/images/Warning.png",
            }
          }
          style={
            Object {
              "marginTop": 14,
              "tintColor": "white",
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "marginHorizontal": 12,
            "marginVertical": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "fontWeight": "600",
              "marginRight": 32,
            }
          }
        >
          Permission Slips!
        </Text>
        <SubTitle>
          Tap to view announcement
        </SubTitle>
      </View>
    </View>
  </TouchableHighlight>
  <TouchableHighlight
    accessibilityLabel="Dismiss Permission Slips!"
    accessibilityTraits="button"
    activeOpacity={0.85}
    onPress={[Function]}
    style={
      Object {
        "padding": 12,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    testID="global-announcement-row.button"
    underlayColor="transparent"
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/images/x-icon.png",
        }
      }
      style={
        Object {
          "tintColor": "#8B969E",
          "width": 16,
        }
      }
    />
  </TouchableHighlight>
</Component>
`;

exports[`GlobalAnnouncementRow renders information announcements 1`] = `
<Component
  contentStyle={
    Array [
      Object {
        "borderColor": "#374A59",
        "borderWidth": 1,
      },
    ]
  }
  hideShadow={true}
>
  <TouchableHighlight
    accessibilityTraits="button"
    activeOpacity={0.85}
    onPress={[Function]}
    testID="global-announcement-row.expand"
    underlayColor="transparent"
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "width": 40,
            },
            Object {
              "backgroundColor": "#374A59",
            },
          ]
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/images/dashboard/info.png",
            }
          }
          style={
            Object {
              "marginTop": 14,
              "tintColor": "white",
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "marginHorizontal": 12,
            "marginVertical": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "fontWeight": "600",
              "marginRight": 32,
            }
          }
        >
          Permission Slips!
        </Text>
        <SubTitle>
          Tap to view announcement
        </SubTitle>
      </View>
    </View>
  </TouchableHighlight>
  <TouchableHighlight
    accessibilityLabel="Dismiss Permission Slips!"
    accessibilityTraits="button"
    activeOpacity={0.85}
    onPress={[Function]}
    style={
      Object {
        "padding": 12,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    testID="global-announcement-row.button"
    underlayColor="transparent"
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/images/x-icon.png",
        }
      }
      style={
        Object {
          "tintColor": "#8B969E",
          "width": 16,
        }
      }
    />
  </TouchableHighlight>
</Component>
`;

exports[`GlobalAnnouncementRow renders question announcements 1`] = `
<Component
  contentStyle={
    Array [
      Object {
        "borderColor": "#374A59",
        "borderWidth": 1,
      },
    ]
  }
  hideShadow={true}
>
  <TouchableHighlight
    accessibilityTraits="button"
    activeOpacity={0.85}
    onPress={[Function]}
    testID="global-announcement-row.expand"
    underlayColor="transparent"
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "width": 40,
            },
            Object {
              "backgroundColor": "#374A59",
            },
          ]
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/images/dashboard/help.png",
            }
          }
          style={
            Object {
              "marginTop": 14,
              "tintColor": "white",
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "marginHorizontal": 12,
            "marginVertical": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "fontWeight": "600",
              "marginRight": 32,
            }
          }
        >
          Permission Slips!
        </Text>
        <SubTitle>
          Tap to view announcement
        </SubTitle>
      </View>
    </View>
  </TouchableHighlight>
  <TouchableHighlight
    accessibilityLabel="Dismiss Permission Slips!"
    accessibilityTraits="button"
    activeOpacity={0.85}
    onPress={[Function]}
    style={
      Object {
        "padding": 12,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    testID="global-announcement-row.button"
    underlayColor="transparent"
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/images/x-icon.png",
        }
      }
      style={
        Object {
          "tintColor": "#8B969E",
          "width": 16,
        }
      }
    />
  </TouchableHighlight>
</Component>
`;

exports[`GlobalAnnouncementRow renders unexpected icon as information 1`] = `
<Component
  contentStyle={
    Array [
      Object {
        "borderColor": "#374A59",
        "borderWidth": 1,
      },
    ]
  }
  hideShadow={true}
>
  <TouchableHighlight
    accessibilityTraits="button"
    activeOpacity={0.85}
    onPress={[Function]}
    testID="global-announcement-row.expand"
    underlayColor="transparent"
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "width": 40,
            },
            Object {
              "backgroundColor": "#374A59",
            },
          ]
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/images/dashboard/info.png",
            }
          }
          style={
            Object {
              "marginTop": 14,
              "tintColor": "white",
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "marginHorizontal": 12,
            "marginVertical": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "fontWeight": "600",
              "marginRight": 32,
            }
          }
        >
          Permission Slips!
        </Text>
        <SubTitle>
          Tap to view announcement
        </SubTitle>
      </View>
    </View>
  </TouchableHighlight>
  <TouchableHighlight
    accessibilityLabel="Dismiss Permission Slips!"
    accessibilityTraits="button"
    activeOpacity={0.85}
    onPress={[Function]}
    style={
      Object {
        "padding": 12,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    testID="global-announcement-row.button"
    underlayColor="transparent"
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/images/x-icon.png",
        }
      }
      style={
        Object {
          "tintColor": "#8B969E",
          "width": 16,
        }
      }
    />
  </TouchableHighlight>
</Component>
`;

exports[`GlobalAnnouncementRow renders warning announcements 1`] = `
<Component
  contentStyle={
    Array [
      Object {
        "borderColor": "#FC5E13",
        "borderWidth": 1,
      },
    ]
  }
  hideShadow={true}
>
  <TouchableHighlight
    accessibilityTraits="button"
    activeOpacity={0.85}
    onPress={[Function]}
    testID="global-announcement-row.expand"
    underlayColor="transparent"
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "width": 40,
            },
            Object {
              "backgroundColor": "#FC5E13",
            },
          ]
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/images/Warning.png",
            }
          }
          style={
            Object {
              "marginTop": 14,
              "tintColor": "white",
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "marginHorizontal": 12,
            "marginVertical": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "fontWeight": "600",
              "marginRight": 32,
            }
          }
        >
          Permission Slips!
        </Text>
        <SubTitle>
          Tap to view announcement
        </SubTitle>
      </View>
    </View>
  </TouchableHighlight>
  <TouchableHighlight
    accessibilityLabel="Dismiss Permission Slips!"
    accessibilityTraits="button"
    activeOpacity={0.85}
    onPress={[Function]}
    style={
      Object {
        "padding": 12,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    testID="global-announcement-row.button"
    underlayColor="transparent"
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/images/x-icon.png",
        }
      }
      style={
        Object {
          "tintColor": "#8B969E",
          "width": 16,
        }
      }
    />
  </TouchableHighlight>
</Component>
`;
